lib/KBaseReport/utils/validation_utils.py

```diff
@@ -93,6 +106,125 @@ def file_or_dir(path):
             raise ValueError(_format_errors(err, f))
 
 
+def valid_dir_path(field, dir_path, error):
+    """ ensure a directory exists """
+    if (not os.path.isdir(dir_path)):
+        error(field, 'does not exist on filesystem')
+
+def valid_file_path(field, file_path, error):
+    """ ensure a file exists """
+    if (not os.path.isfile(file_path)):
+        error(field, 'does not exist on filesystem')
+
+
+def valid_json(field, string, error):
+    try:
+        template_data = json.loads(string)
+    except JSONDecodeError as err:
+        error(field, 'Invalid JSON: ' + err.msg + ' ' + str(err.pos))
+
+
+def validate_template_params(params, config, with_output_file=False):
+    """ Validate all parameters to KBaseReportImpl#render_template
+
+    :param params:  (dict)  input to be validated
+    :param config:  (dict)  app config
+    :param with_output_file: (bool) whether or not the output_file param should be validated
+
+    :return:
+    params (dict) - validated params
+    """
+    validator = Validator(purge_unknown = True)
+
+    config_validation_schema = {
+        'scratch': {
+            'type': 'string',
+            'minlength': 2,
+            'required': True,
+            'validator': valid_dir_path,
+        },
+        'template_toolkit': {
+            'type': 'dict',
+        }
+    }
+
+    if not validator.validate(config, config_validation_schema):
+        raise TypeError(_format_errors(validator.errors, config))
+
+    validated_params = {
+        'scratch': config['scratch'],
+        'template_config': config['template_toolkit'] if 'template_toolkit' in config else {}
+    }
+
+    scratch_path = config['scratch']
+
+    def path_contains_scratch(field, file_path, error):
+
```

Comment:
Minor, but can you remove this linebreak so it's easier to see that the below conditional is part of this function body